.github/workflows/dependencies/updater.py

```diff
@@ -14,437 +19,551 @@
 # Dry run flag
 DRY_RUN = os.environ.get("DRY_RUN", "0") == "1"
 
-import timeit
+# utils for tag comparison
+BASEVERSION = re.compile(
+    r"""[vV]?
+    (?P<major>0|[1-9]\d*)
+    (\.
+    (?P<minor>0|[1-9]\d*)
+    (\.
+    (?P<patch>0|[1-9]\d*)
+    )?
+    )?
+    """,
+    re.VERBOSE,
+)
+
+
+def coerce(version: str) -> Optional[Version]:
+    match = BASEVERSION.search(version)
+    if not match:
+        return None
+
+    ver = {
```

Comment:
I'd add a comment here about what this is doing, including when using this variable on the `Version(**ver)` line. Looks complex enough to warrant it.